controllers/kubeadmconfig_controller_test.go

```diff
@@ -192,6 +192,34 @@ func TestKubeadmConfigReconciler_Reconcile_ReturnEarlyIfMachineHasBootstrapData(
 	}
 }
 
+// Return early If the owning machine does not have a associated cluster
+func TestKubeadmConfigReconciler_Reconcile_ReturnEarlyIfMachineHasNoCluster(t *testing.T) {
```

Comment:
@chuckha @ncdc based on the discussions from the f2f planning, is this the behavior that we actually want? Or do we want to allow a Machine without a Cluster as long as the label isn't set and have some fallback behavior to just not provide the defaults we set from the Cluster resource when that is the case?

Reply:
This is for the use-case that only uses the Machine API?

Reply:
@chuckha indeed

Reply:
My understanding was that a Cluster object was still required even if it's an empty object. It's good to figure this out now though. Is that also your understanding?

Reply:
That is the direction that @ncdc, @vincepri, and myself have been advocating for, but I believe that the Cluster has been specified to still be optional at this point in time.

Reply:
Cluster is required as of today. This is not a change in behavior but a formalization of the current behavior. If we can open an issue to get the Machine API use-case here then we can work on iterating from this place:

[CABPK requires a Cluster](https://github.com/kubernetes-sigs/cluster-api-bootstrap-provider-kubeadm/blob/c94ead1dbfc947ed7625d2eb3c8b4183ac7efd62/controllers/kubeadmconfig_controller.go#L128)
[Cluster API requires a Cluster](https://github.com/kubernetes-sigs/cluster-api/blob/2c4cbfa482d3bc7857333aeb1a0eb0599e2320bd/controllers/machine_controller.go#L113)
[CAPA requires a Cluster](https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/256276cd93bb38d4a1d6dae98d376908689c9d5d/controllers/awsmachine_controller.go#L91)